### DISCUSSION TITLE
Coordinate system differences right/left

### DISCUSSION THREAD

**Discussion by stiangglanda** [assimp/assimp#4156] [Q&A] [answered]
Hi i am currently implementing skeletal animation using directx and i use a left handed coordinate system 

I have two questions first since i need to transpose every matrix (bone offset in my case) do i need to do something to rotation vector 4, position vector 3 and scaling vector 3 maybe swap y and z?

And second I don't know what the transformation in aiNode does i thought it meight be(for the bones) the bone offset but since the data isn't the same and it is not just bones this can't be true or is it scaling x position x rotation already calculated?

**Comment by stiangglanda** [comment, marked as answer]
I have fixed it my calculation was wrong and the second assumption (scal*pos*rot) was right

**Comment by kimkulling** [comment]
Great, thanks a lot for oyur update!
